### PR TITLE
754 add postgres dataset entry repository and rebac repository implementations

### PR DESCRIPTION
**Pull request by zaychenko-sergei**
## Description

Related issue #754 

## Checklist before requesting a review

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [ ] Observability:
    
  - [ ] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ✅
    
  - [ ] Alerts: ✅
- [ ] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [ ] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅
    
  - [ ] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ✅
    
  - [ ] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1)

### CODE REVIEW

File: migrations/sqlite/20240814131142_auth_rebac_tables.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 CREATE TABLE auth_rebac_properties
 (
-    entity_type    VARCHAR(25)  NOT NULL,
+    entity_type    VARCHAR(25)  CHECK (entity_type IN ('account', 'dataset')) NOT NULL,
```
Comment (s373r) [inline migrations/sqlite/20240814131142_auth_rebac_tables.sql:5]: Note: yes, that's a good point

File: src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        let stack_dataset_id = dataset_id.as_did_str().to_stack_string();
+
+        let update_result = sqlx::query!(
+            r#"
+            UPDATE dataset_entries
+                SET dataset_name = $2
+                WHERE dataset_id = $1
+            "#,
+            stack_dataset_id.as_str(),
+            new_name.as_str(),
```
Comment (s373r) [inline src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs:205]: Minor: reorder, so as not to catch the eye
```sql 
SET dataset_name = $1
WHERE dataset_id = $1
```

**Review by s373r — APPROVED** [review]
LGTM + a minor optional comment

File: src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs
Comment (s373r) [inline src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs:1]: Please register added repos in `configure_database_components()`

File: src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs (hunk 2 above)
Comment (zaychenko-sergei) [reply src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs:205]: Corrected here and in SQlite original

File: src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs
Comment (zaychenko-sergei) [reply src/infra/datasets/postgres/src/repos/postgres_dataset_entry_repository.rs:1]: Done

**Pull request merged**